api/napi/src/interpreter/component_instance.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use i_slint_compiler::langtype::Type;
+use napi::{Env, Error, JsFunction, JsUnknown, NapiRaw, NapiValue, Ref, Result};
+use slint_interpreter::{ComponentHandle, ComponentInstance, Value};
+
+use super::JsComponentDefinition;
+
+#[napi(js_name = "ComponentInstance")]
+pub struct JsComponentInstance {
+    inner: ComponentInstance,
+}
+
+impl From<ComponentInstance> for JsComponentInstance {
+    fn from(instance: ComponentInstance) -> Self {
+        Self { inner: instance }
+    }
+}
+
+#[napi]
+impl JsComponentInstance {
+    #[napi(constructor)]
+    pub fn new() -> napi::Result<Self> {
+        Err(napi::Error::from_reason(
+            "ComponentInstance can only be created by using ComponentCompiler.".to_string(),
+        ))
+    }
+
+    #[napi]
+    pub fn definition(&self) -> JsComponentDefinition {
+        self.inner.definition().into()
+    }
+
+    #[napi]
+    pub fn run(&self) {
+        self.inner.run().unwrap()
+    }
+
+    #[napi]
+    pub fn get_property(&self, env: Env, name: String) -> Result<JsUnknown> {
+        let value = self
+            .inner
+            .get_property(name.as_ref())
+            .map_err(|e| Error::from_reason(e.to_string()))?;
+        super::value::to_js_unknown(&env, &value)
+    }
+
+    #[napi]
+    pub fn set_property(&self, env: Env, prop_name: String, js_value: JsUnknown) -> Result<()> {
+        let ty = self
+            .inner
+            .definition()
+            .properties_and_callbacks()
+            .find_map(|(name, proptype)| if name == prop_name { Some(proptype) } else { None })
+            .ok_or(())
+            .map_err(|_| napi::Error::from_reason("Cannot read slint type of property."))?;
```

Comment:
Please add a test for this, i.e. the case when calling `set_property` on a non-existent property.

(same for set_callback, etc.)